cmake/onnxruntime_providers.cmake

```diff
@@ -1074,11 +1078,17 @@ if (onnxruntime_USE_DML)
   endfunction()
 
   target_add_dml(onnxruntime_providers_dml)
-  target_link_libraries(onnxruntime_providers_dml PRIVATE d3d12.lib dxgi.lib)
+  if (GDK_PLATFORM STREQUAL Scarlett)
+    target_link_libraries(onnxruntime_providers_dml PRIVATE ${gdk_dx_libs})
+  else()
+    target_link_libraries(onnxruntime_providers_dml PRIVATE d3d12.lib dxgi.lib)
+  endif()
 
   target_link_libraries(onnxruntime_providers_dml PRIVATE delayimp.lib)
 
-  set(onnxruntime_DELAYLOAD_FLAGS "${onnxruntime_DELAYLOAD_FLAGS} /DELAYLOAD:DirectML.dll /DELAYLOAD:d3d12.dll /DELAYLOAD:dxgi.dll /DELAYLOAD:api-ms-win-core-com-l1-1-0.dll /DELAYLOAD:shlwapi.dll /DELAYLOAD:oleaut32.dll /ignore:4199")
+  if (NOT GDK_PLATFORM)
+    set(onnxruntime_DELAYLOAD_FLAGS "${onnxruntime_DELAYLOAD_FLAGS} /DELAYLOAD:DirectML.dll /DELAYLOAD:d3d12.dll /DELAYLOAD:dxgi.dll /DELAYLOAD:api-ms-win-core-com-l1-1-0.dll /DELAYLOAD:shlwapi.dll /DELAYLOAD:oleaut32.dll /ignore:4199")
```

Comment:
Just curious, is delay loading not supported here?

Reply:
We just don't want to pick up these libraries: d3d12.dll, dxgi.dll, shlwapi.dll, and oleaut32.dll won't exist. I think this might compile with a warning like "LINK : warning LNK4199: /DELAYLOAD:d3d12.dll ignored; no imports found from d3d12.dll", but I'd prefer the build fail if we inadvertently link with a symbol from most of these libs.